daemon/daemon.go

```diff
@@ -786,6 +788,31 @@ func NewDaemon(config *Config, eng *engine.Engine) (*Daemon, error) {
 }
 
 func NewDaemonFromDirectory(config *Config, eng *engine.Engine) (*Daemon, error) {
+	// register handler for handling dead children
+	log.Error("alsknaflkndklfndlsfnnsdl")
+	log.Error(os.Getpid())
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGCHLD)
+	go func() {
```

Comment:
Can you move the anonymous function to a top level one and just do `go foo()`.  Also check for PID 1 again